lib/middleware/anonymous_cache.rb

```diff
@@ -11,6 +11,17 @@ def self.anon_cache(env, duration)
       env["ANON_CACHE_DURATION"] = duration
     end
 
+    # This gives us an API to insert anonymous cache segments
+    def self.cache_key_segments
+      @@cache_key_segments ||= {
+        m: ->(h) { h.is_mobile? },
+        c: ->(h) { h.is_crawler? },
+        b: ->(h) { h.has_brotli? },
+        t: ->(h) { h.theme_ids.join(',') },
+        ca: ->(_) { GlobalSetting.compress_anon_cache.to_s }
```

Comment:
Note: previously the `compress_anon_cache` was not separated by a variable and simply appended `true` or `false` to the key, which I assume was a mistake and not deliberate.

Reply:
GlobalSetting.compress_anon_cache.to_s is never going to change at runtime, maybe just hardcode it?

One note here is that the new flexible implementation is almost 3x slower than the original per: https://gist.github.com/00a759f5d8bc45726773afb1dbde379a 

I am normally not such a stickler, but this is a code path that every single request goes through so I am a bit sensitive.

Maybe we keep the old method for the base and then allow people to add stuff to "additional_cache_key_segments" if they wish to. As an advantage here they will not be able to break base caching support in a plugin cause they will only be allowed to add stuff on.